test/parallel/test-stdio-undestroy.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const spawn = require('child_process').spawn;
+const fs = require('fs');
+const fixtures = require('../common/fixtures');
+
+if (process.argv[2] === 'child') {
+  process.stdout.destroy()
+  console.log('out')
+  console.error('err')
+  return;
+}
+
+// Run the script in a shell but close stdout and stderr.
+const cmd = `"${process.execPath}" "${__filename}" child`;
+const proc = spawn('/bin/sh', ['-c', cmd], { stdio: 'pipe' });
```

Comment:
The comment seems to be inaccurate, and using a shell here doesn’t make a difference except make the test less reliable?

```suggestion
const proc = spawn(process.execPath, [__filename, 'child'], { stdio: 'pipe' });
```